functors_task_1/student/include/impl.h

```diff
@@ -40,9 +51,17 @@ struct MixingPolicy
      * Whiskey + SevenUp = SevenPlusSeven;
      * Others + Others = Oops;
      */
-    static Cocktail mix(/**???*/)
+    static Cocktail mix(AlcoholDrink alcDrink, NonAlcoholDrink nonAlcDrink)
     {
+        static Cocktail mix(AlcoholDrink alcohol, NonAlcoholDrink nonAlcohol)
+            {
+                if(alcohol == AlcoholDrink::Gin && nonAlcohol == NonAlcoholDrink::LimeJuice) return Cocktail::Gimlet;
```

Comment:
Please consider https://isocpp.github.io/CppCoreGuidelines/CppCoreGuidelines#es70-prefer-a-switch-statement-to-an-if-statement-when-there-is-a-choice